src/Storage/StorageObject.php

```diff
@@ -219,6 +219,12 @@ public function delete(array $options = [])
     public function update(array $metadata, array $options = [])
     {
         $options += $metadata;
+
+        // can only set predefinedAcl or acl
+        if (isset($options['predefinedAcl'])) {
+            $options['acl'] = null;
```

Comment:
It isn't just this request they conflict on, we need to actually wipe the value stored upstream in order to be able to set predefinedAcl.